src/Task.js

```diff
@@ -57,7 +57,6 @@ class Task {
 
             taskInfo.parameters.forEach((param) => {
               parameters[param.name] = param;
-              delete parameters[param.name].name;
```

Comment:
We were removing the name property because it is the key as well.  We have found that it makes more sense to keep it in the object as well, that way consumers don't have to pass around a name string separately.

The taskInfo object used to look like this:
{
    "name": "SpectralIndex",
    "displayName": "Spectral Index",
    "description": "This task creates a spectral index raster from one pre-defined spectral index. Spectral indices are combinations of surface reflectance at two or more wavelengths that indicate relative abundance of features of interest.",
    "parameters": {
        INPUT_RASTER: {
            "displayName": "Input Raster",
            "dataType": "ENVIRASTER",
            "description": "Specify a raster from which to generate a spectral index raster.",
            "direction": "INPUT",
            "parameterType": "required"
        },
    }
}

But now it looks like this:
{
    "name": "SpectralIndex",
    "displayName": "Spectral Index",
    "description": "This task creates a spectral index raster from one pre-defined spectral index. Spectral indices are combinations of surface reflectance at two or more wavelengths that indicate relative abundance of features of interest.",
    "parameters": {
        INPUT_RASTER: {
            **"name": "INPUT_RASTER",**
            "displayName": "Input Raster",
            "dataType": "ENVIRASTER",
            "description": "Specify a raster from which to generate a spectral index raster.",
            "direction": "INPUT",
            "parameterType": "required"
        },
    }
}